frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -306,7 +306,7 @@ function FileManagerMenu:onShowMenu()
     }
 
     local main_menu
-    if Device:isTouchDevice() then
+    if Device:isTouchDevice() or Device:hasKeys() then
```

Comment:
When will we reach the else branch? What's the intent behind this change exactly? Is this saying the else branch and presumably therefore also `Menu:itemTableFromTouchMenu()` is deprecated?

Reply:
There is also a `hasDPad()`. I dunno if we can have device that have this, but not hasKeys() neither isTouchDevice().
May be we need some kind of reference about devices capabilities, and their available physical keys and to what they map to.

Reply:
I see, but if hasKeys can go up/down/left/right then hasDPad should be able to do that even better. :-P

Then there's hasKeyboard but in practice that seems to be additive.